lib/main_common.pm

```diff
@@ -1110,7 +1110,7 @@ sub load_consoletests {
         }
     }
     loadtest 'console/install_all_from_repository' if get_var('INSTALL_ALL_REPO');
-    if (check_var_array('SCC_ADDONS', 'tcm') && get_var('PATTERNS') && sle_version_at_least('12-SP3')) {
+    if (check_var_array('SCC_ADDONS', 'tcm') && get_var('PATTERNS') && is_sle('<15') && !get_var("MEDIA_UPGRADE")) {
```

Comment:
with this PR you also exclude deregister from SLE 15 and add it for all other SLE versions lower SLE 15 . Not say that it is wrong per ce but can you explain your motivation ?

Reply:
@dehai I think you have read the `sle_version_at_least('12-SP3')` the wrong way around. Should be `is_sle('12-SP3+')`, isn't it?

Reply:
@okurz @asmorodskyi because SLE15 does not have tool chain module, it has integrated into development tools module.

Reply:
@dehai and what about SLE 12 SP1 and SLE 12 SP2 ? are you ok to deregister tcm there ?

Reply:
@dehai, I don't this it is reasonable solution here. We don't have tcm in SLE15, but we do have deregister function in SLE15. In this case, we would never run `SUSEConnect -d` to dergister modules in SLE15 again.

Reply:
@asmorodskyi I think it's okay to deregister tcm on SLE12 SP1/SP2, this test module is for testing  SUSEConnect feature on deregister one module which has already registered. this module maintainer is Xu Yi, she checked a certain module which is tcm. And, I think this test module should not be tested on migration job group, it should be tested on functional . Because we focus on migration testing, not a functional feature testing. Thanks!

Reply:
@Jiawei517 Yes, I agree with you, SLE15 does not have tcm anymore. But , in this test module, script  use tcm as certain module to test this feature. and we do not need to test this module in our migration job group. I think it should be covered by functional. So, skip it.

Reply:
you convince me :) . LGTM from me . But I will let you continue discussion with @Jiawei517  and @okurz

Reply:
Please put something into the code. The code should be self a explanatory, if not at least describe the reason in a comment

Reply:
@okurz  added reason on comment.

Reply:
Sorry, I don't see any updated source code which would include the comment.

Reply:
@okurz I have rebased my PR, and updated reasons on commit. I don't add this long explanatory into code because this is main_common module, if added, it will be very ugly.

Reply:
@dehai I think you do better add the comment to `lib/main_common.pm`